wagtail/snippets/tests/test_viewset.py

```diff
@@ -413,7 +413,7 @@ def test_custom_chooser_pagination(self):
 
         # FullFeaturedSnippet is set to display 15 per page
         self.assertEqual(FullFeaturedSnippet.objects.all().count(), 32)
-        self.assertContains(response, "Page 1 of 3")
+        # self.assertContains(response, "Page 1 of 3")
```

Comment:
Previously, the `per_page` attribute was overridden by the `ChooserViewSet`. However, I've now updated the attribute name, removing the possibility of an override. Consequently, we won't be able to use `chooser_per_page` in the `SnippetViewset` to update this value, as the `per_page` value of `ChooserViewSet` no longer matters. Therefore, I've updated this test accordingly.

Please correct me if I'm mistaken.

Reply:
I don't think this is right - we _do_ want people to be able to override [the per_page attribute through ChooserViewSet](https://docs.wagtail.org/en/stable/reference/viewsets.html#wagtail.admin.viewsets.chooser.ChooserViewSet.per_page), and via [`chooser_per_page` on SnippetViewset](https://docs.wagtail.org/en/stable/reference/viewsets.html#wagtail.snippets.views.snippets.SnippetViewSet.chooser_per_page), as these are both documented APIs that people are actively making use of.

I think the correct fix is either (or both) of:
* Update `wagtail.images.views.chooser` so that `WAGTAILIMAGES_CHOOSER_PAGE_SIZE` is passed as part of `ImageChooserViewSet` instead of `BaseImageChooseView`;
* Change the behaviour of `ChooserViewSet.per_page` and `SnippetViewSet.chooser_per_page` so that in the base viewset classes, they're undefined by default, and we _only_ pass those values on to the view if they've been explicitly defined on the viewset. That way, if the developer _doesn't_ set it at the viewset level, it remains unchanged from whatever is set on the view.

I think the latter is going to be hard to do in a way that preserves backward compatibility (especially as I think we support setting `per_page = None` to indicate "no pagination"), so to keep things simple I'd probably go with the first option.

Reply:
Thanks for the input, I agree, going with the first idea seems like the best approach to fix the image chooser issue. And from what I know, users don't really  have any option to change `per_page` value in `ChooseView`, so probably there shouldn't be a big issue with conflicting attribute names. Plus, we already let users set the number of items per page for all the viewsets.

So, sticking with the first option sounds like the best way to go!